airflow/jobs/scheduler_job.py

```diff
@@ -717,7 +718,10 @@ def _process_task_instances(self, dag, task_instances_list, session=None):
             run.dag = dag
             # todo: preferably the integrity check happens at dag collection time
             run.verify_integrity(session=session)
-            run.update_state(session=session)
+            finished_tasks = run.get_task_instances(state=State.finished() + [State.UPSTREAM_FAILED],
+                                                    session=session)
```

Comment:
This works, but it asks for a lot more columns and rows than we need.

We could try changing the return inside this function from `return tis.all()` to just `return tis`, and this line could become:

```python
            finished_tasks = run.get_task_instances(state=State.finished() + [State.UPSTREAM_FAILED],
                                                    session=session).options(load_only("task_id", "state"))
```

https://docs.sqlalchemy.org/en/13/orm/loading_columns.html#load-only-and-wildcard-options

Do you think this is worth it or not worth it?

Reply:
Yes I think it worth it, this query happens a lot..